### PR TITLE
Add postGis, Add tests for python files

### PR DESCRIPTION
**Pull request by henrykironde**
1. This add postgis data processing.
2. Adds test for python custom files by registering the files.
3. Refactor code.

### CODE REVIEW

**Comment by henrykironde** [comment]
ref : #1111 ,#1129, #1154

**Comment by henrykironde** [comment]
[WIP]: I have to revisit and refine the way files close including `def disconnect_files(self)` and add tests for spatial datasets. Any comments, reviews are all welcome.

**Comment by henrykironde** [comment]
ToDOs: 
- [ ] Walk through the PR one more time.
- [ ] Test exceptions. As Ethan stated, `The error handling doesn't seem to be quite right here. I ran into the following while testing:`, This could be because of the BaseException. I will test it.

File: retriever/engines/msaccess.py [hunk 1]
```diff
@@ -135,24 +136,13 @@ def insert_data_from_file(self, filename):
...
-        """Determine if the table already exists in the database"""
-        if not hasattr(self, 'existing_table_names'):
-            self.existing_table_names = set()
-            for row in self.cursor.tables():
-                tableinfo = row[2]
-                if not tableinfo.startswith("MSys"):
-                    # ignore system tables
-                    database, table = tableinfo.split()
-                    self.existing_table_names.add((database, table))
-        return self.table_name(name=tablename, dbname=dbname).lower() in self.existing_table_names
```
[the remark this answers is not in the document]
Comment (henrykironde) [reply retriever/engines/msaccess.py]: So table_exist is a function that is not used. 
If we ever decide to use the function, we shall have to change procedures defined in the functions.

File: retriever/engines/postgres.py [hunk 2]
```diff
@@ -52,12 +70,27 @@ def create_db(self):
...
+        PostgreSQL needs to commit operations individually.
+        Enable PostGis extensions if a script has a non tabular table.
+        """
+        if self.table and self.table.dataset_type and \
+                not self.table.dataset_type == "TabularDataset":
+            try:
+                # Check if Postgis is installed and EXTENSION are Loaded
+                if self.execute("SELECT PostGIS_full_version();") or \
+                        self.execute("SELECT PostGIS_version()"):
+                    pass
```
[the remark this answers is not in the document]
Comment (henrykironde) [reply retriever/engines/postgres.py]: The raise was printing out 
```bash
function postgis_full_version() does not exist
LINE 1: SELECT PostGIS_full_version();
```
I am not using exit()

File: retriever/lib/engine.py [hunk 3]
```diff
@@ -769,6 +796,10 @@ def set_table_delimiter(self, file_path):
         self.auto_get_delimiter(dataset_file.readline())
         dataset_file.close()
 
+    def supported_raster(self, path, ext=None):
```
[the remark this answers is not in the document]
Comment (henrykironde) [reply retriever/lib/engine.py]: doc string added.

**Comment by henrykironde** [comment]
@ethanwhite, this is ready for a review.

**Comment by ethanwhite** [comment]
Very cool @henrykironde. Great to have this in!

**Pull request merged**